MultiServer.py

```diff
@@ -1903,27 +1903,29 @@ async def process_client_cmd(ctx: Context, client: Client, args: dict):
         
         elif cmd == 'UpdateHint':
             location = args["location"]
-            player = args["player"]
-            status = args["status"]
-            if not isinstance(player, int) or not isinstance(location, int) \
-                    or (status is not None and not isinstance(status, int)):
+            location_player = args["player"]
+            create_if_not_exists = args.get("create_if_not_exists", False)
+            status_int = args.get("status")
+
+            if not isinstance(location_player, int) or not isinstance(location, int) \
+                    or (status_int is not None and not isinstance(status_int, int)):
                 await ctx.send_msgs(client,
                                     [{'cmd': 'InvalidPacket', "type": "arguments", "text": 'UpdateHint',
                                       "original_cmd": cmd}])
                 return
-            hint = ctx.get_hint(client.team, player, location)
-            if not hint:
-                return  # Ignored safely
-            if hint.receiving_player != client.slot:
-                await ctx.send_msgs(client,
-                                    [{'cmd': 'InvalidPacket', "type": "arguments", "text": 'UpdateHint: No Permission',
-                                      "original_cmd": cmd}])
-                return
-            new_hint = hint
-            if status is None:
-                return
+
+            hint = ctx.get_hint(client.team, location_player, location)
+
+            if status_int is None:
+                if hint is None:
+                    # New hints are created with unspecified by default
+                    status_int = HintStatus.HINT_UNSPECIFIED
+                else:
+                    # If the hint already exists and no status was provided, there is no point to the packet
+                    return
+
             try:
-                status = HintStatus(status)
+                status = HintStatus(status_int)
```

Comment:
I would move the  try/catch for `status = HintStatus(status_int)` above the `hint = ctx.get_hint(client.team, location_player, location)` as it breaks the flow that there is a check for if hint is None right below it